server2/Arcadia.Assistant/Arcadia.Assistant.AppCenterBuilds/AppCenterBuilds.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.Collections.Generic;
+using System.Fabric;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using Arcadia.Assistant.AppCenterBuilds.Contracts;
+using Arcadia.Assistant.AppCenterBuilds.Contracts.AppCenter;
+using Arcadia.Assistant.AppCenterBuilds.Contracts.Interfaces;
+using Arcadia.Assistant.MobileBuild.Contracts.Interfaces;
+using Microsoft.Extensions.Logging;
+using Microsoft.ServiceFabric.Services.Communication.Runtime;
+using Microsoft.ServiceFabric.Services.Runtime;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace Arcadia.Assistant.AppCenterBuilds
+{
+    /// <summary>
+    /// An instance of this class is created for each service instance by the Service Fabric runtime.
+    /// </summary>
+    public class AppCenterBuilds : StatelessService
+    {
+        private readonly IDownloadApplicationSettings configuration;
+        private readonly IHttpClientFactory clientFactory;
+        private readonly IMobileBuildActorFactory mobileBuildFactory;
+
+        public AppCenterBuilds(StatelessServiceContext context, IDownloadApplicationSettings configuration, IHttpClientFactory clientFactory, IMobileBuildActorFactory mobileBuildFactory)
+            : base(context)
+        {
+            this.configuration = configuration;
+            this.clientFactory = clientFactory;
+            this.mobileBuildFactory = mobileBuildFactory;
+        }
+
+        /// <summary>
+        /// Optional override to create listeners (e.g., TCP, HTTP) for this service replica to handle client or user requests.
+        /// </summary>
+        /// <returns>A collection of listeners.</returns>
+        protected override IEnumerable<ServiceInstanceListener> CreateServiceInstanceListeners()
+        {
+            return new ServiceInstanceListener[0];
+        }
+
+        /// <summary>
+        /// This is the main entry point for your service instance.
+        /// </summary>
+        /// <param name="cancellationToken">Canceled when Service Fabric needs to shut down this service instance.</param>
+        protected override async Task RunAsync(CancellationToken cancellationToken)
+        {
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                ServiceEventSource.Current.ServiceMessage(this.Context, "Request build version");
+
+                if (this.configuration != null)
+                {
+                    // for android
+                    var result = await UpdateMobileBuild(ApplicationTypeEnum.Android, this.configuration?.AndroidGetBuildsUrl ?? string.Empty, this.configuration?.AndroidGetBuildDownloadLinkTemplateUrl ?? string.Empty, cancellationToken);
```

Comment:
how configuration / these fields can be null? I'd just check for null or empty and did nothing in that case, rather than failing with exception

Reply:
New additional checking added.